test/integration/render_task_test.rb

```diff
@@ -0,0 +1,191 @@
+    fixture = 'deployment.yaml.erb'
+
+    assert_render_success(render.run([fixture], test_output_stream))
+    stdout_assertion do |output|
+      assert_equal output, <<~RENDERED
+        ---
+        apiVersion: extensions/v1beta1
+        kind: Deployment
+        metadata:
+          name: web
+        spec:
+          replicas: 1
+          template:
+            metadata:
+              labels:
+                name: web
+                app: test-partials
+            spec:
+              containers:
+              - name: sleepy-guy
+                image: busybox
+                imagePullPolicy: IfNotPresent
+                command:
+                - sleep
+                - '8000'
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod1
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod1
+            image: busybox
+            args:
+            - echo
+            - log from pod1
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod1
+        data:
+          supports_partials: yep
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: independent-configmap
+        data:
+          value: renderer test
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod2
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod2
+            image: busybox
+            args:
+            - echo
+            - log from pod2
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod2
+        data:
+          supports_partials: yep
+      RENDERED
+    end
+  end
+
+  def test_render_task_rendering_all_files
+    render = build_render_task(fixture_path('hello-cloud'))
+
+    assert render.run([], test_output_stream)
+    stdout_assertion do |output|
+      assert_match(/name: bare-replica-set/, output)
+      assert_match(/name: hello-cloud-configmap-data/, output)
+      assert_match(/name: ds-app/, output)
+      assert_match(/kind: PodDisruptionBudget/, output)
+      assert_match(/name: hello-job/, output)
+      assert_match(/name: redis/, output)
+      assert_match(/name: role-binding/, output)
+      assert_match(/name: resource-quotas/, output)
+      assert_match(/name: build-robot/, output)
+      assert_match(/name: stateful-busybox/, output)
+      assert_match(/name: hello-cloud-template-runner/, output)
+      assert_match(/name: unmanaged-pod-\w+/, output)
+      assert_match(/name: web/, output)
+    end
+  end
+
+  def test_render_invalid_binding
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a', 'b': 'binding-b')
+    fixture = 'deployment.yaml.erb'
+
+    assert !render.run([fixture], test_output_stream)
```

Comment:
I'd like to have an `assert_render_failure` for this that also checks that the stderr logging reports the failure correctly.